### PR TITLE
fix: Resolve issues in environments with frozen Symbol

### PR DESCRIPTION
**Pull request by benlesh**
- Also adds a test for frozen Symbol.

Related #47

### CODE REVIEW

File: es/ponyfill.js [hunk 1]
```diff
@@ -6,8 +6,18 @@ export default function symbolObservablePonyfill(root) {
 		if (Symbol.observable) {
 			result = Symbol.observable;
 		} else {
-			result = Symbol('observable');
-			Symbol.observable = result;
+
+			// This just needs to be something that won't trample other user's Symbol.for use
+			// It also will guide people to the source of their issues, if this is problematic.
+			// META: It's a resource locator!
+			result = Symbol.for('https://github.com/benlesh/symbol-observable');
```
Comment (erights) [inline es/ponyfill.js:13]: Cool! Very RDF

File: test/ponyfill.js [hunk 2]
```diff
@@ -23,19 +23,53 @@ describe('ponyfill unit tests', function () {
...
+  }
+
+  var forLookup = {};
+
+  FakeSymbol.for = function (id) {
+    if (!forLookup[id]) {
+      forLookup[id] = FakeSymbol(id);
+    }
+    return forLookup[id];
+  };
```
Comment (erights) [inline test/ponyfill.js:72]: This seems more complex that necessary. To emulate `Symbol.for`, just avoid a counting prefix. Return the same string each time with the same fixed prefix. No state needed.

**Review by erights — APPROVED** [review]
LGTM. Thanks!

**Review by kriskowal — APPROVED** [review]

**Comment by benlesh** [comment]
I will release this as a full major version change, as it is possible that there are esoteric run times that have `Symbol` but not `Symbol.for`, and this may break them.

I'm not certain that even exist though, so I'm not going to fix it in this release, as they can stick with a previous release and they would be fine. I will be sure to try to note that in the changelog.

I will merge this and release tomorrow morning.

**Comment by webmaster128** [comment]
Thank you for this patch.

> I will release this as a full major version change, as it is possible that there are esoteric run times that have `Symbol` but not `Symbol.for`, and this may break them.

According to caniuse.com, [Symbol](https://www.caniuse.com/mdn-javascript_builtins_symbol) and [Symbol.for](https://www.caniuse.com/mdn-javascript_builtins_symbol_for) have practically the same support. The only difference I see is in Chrome 38-39 (released in 2014) and Opera 25-26 (released in 2014).

**Pull request merged**